### PR TITLE
Add support for logging in as others using django-loginas

### PR DESCRIPTION
**Pull request by hodgestar**

### CODE REVIEW

**Comment by hodgestar** [comment]
See https://github.com/stochastic-technologies/django-loginas#installing-django-loginas for loginas setup instructions. We'll want to try this on staging before landing.

**Comment by hodgestar** [comment]
We should consider whether we want to customize the `CAN_LOGIN_AS` function to, e.g., restrict who one can login as as well as which users can use the feature.

**Comment by hodgestar** [comment]
Ready for review.

**Comment by jerith** [comment]
This needs some tests to make sure we've set it up properly and aren't leaking access to people that shouldn't have it.

**Comment by hodgestar** [comment]
Any thoughts on what the tests should look like? Calls to the loginas view and assertions of success / failure?

**Comment by jerith** [comment]
Something like that, I'd guess.

**Comment by hodgestar** [comment]
Ready for re-review.

File: go/tests/test_urls.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        self.user_helper_2 = self.vumi_helper.make_django_user(
+            email='user2@example.com')
+
+        self.superuser_client = self.vumi_helper.get_client(
+            username='superuser@example.com')
+        self.user_client_1 = self.vumi_helper.get_client(
+            username='user1@example.com')
+
+    def test_successful_login_as(self):
+        """ Superusers should be able to use login-as. """
```
Comment (jerith) [inline go/tests/test_urls.py]: Nitpick: Can we use multiline docstrings (with the quotes on their own lines) even when the entire docstring fits on a single line?

File: go/tests/test_urls.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+class TestLoginAs(GoDjangoTestCase):
+
+    def setUp(self):
+        self.vumi_helper = self.add_helper(DjangoVumiApiHelper())
+        self.superuser_helper = self.vumi_helper.make_django_user(
+            superuser=True, email='superuser@example.com')
+        self.user_helper_1 = self.vumi_helper.make_django_user(
+            email='user1@example.com')
+        self.user_helper_2 = self.vumi_helper.make_django_user(
+            email='user2@example.com')
```
Comment (jerith) [inline go/tests/test_urls.py]: Can we use better names than "user1" and "user2" here? Maybe "normal_user" and "loginas_target"?

**Comment by jerith** [comment]
Thanks for the tests. I'd like us (maybe @badlard?) to bang on this on staging a bit before landing to make sure we're not missing anything obvious, but I'm happyish with the changes now that they're tested.

**Comment by hodgestar** [comment]
Agreed about staging. Ready for re-re-review.

**Comment by jerith** [comment]
:+1:

**Pull request merged**